sklearn/utils/__init__.py

```diff
@@ -149,7 +149,13 @@ def safe_indexing(X, indices):
     """
     if hasattr(X, "iloc"):
         # Pandas Dataframes and Series
-        return X.iloc[indices]
+        try:
+            return X.iloc[indices]
+        except ValueError:
+            # This is due to cython not allowing readonly buffers
```

Comment:
I would be more precise, e.g.:

Cython typed memoryviews internally used in pandas do not support readonly buffers.